Developing with webpack and source maps not supported

As a developer, I want to use webpack with `devtool` setting that generates separate source maps. Flow crashes when trying to append the bundle because of some assumptions about the stats json.

To reproduce, download `skeleton-starter-flow` and add the following under the `jetty-maven-plugin` `<configuration>` section (or set the property in some of the other supported ways):
```
  <systemProperties>
      <systemProperty>
          <name>vaadin.devmode.webpack.options</name>
          <value>-d --devtool cheap-module-source-map</value>
      </systemProperty>
  </systemProperties>
```
Start the server in dev mode with `mvn jetty:run`. When visiting the application in the browser, the following exception is raised:
```
javax.servlet.ServletException: javax.servlet.ServletException: com.vaadin.flow.server.ServiceException: java.lang.ClassCastException: elemental.json.impl.JreJsonArray cannot be cast to elemental.json.JsonString
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:168)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:505)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:370)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:267)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:781)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:917)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.servlet.ServletException: com.vaadin.flow.server.ServiceException: java.lang.ClassCastException: elemental.json.impl.JreJsonArray cannot be cast to elemental.json.JsonString
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:249)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:876)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:214)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1711)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1347)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1678)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1249)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:152)
	... 10 more
Caused by: com.vaadin.flow.server.ServiceException: java.lang.ClassCastException: elemental.json.impl.JreJsonArray cannot be cast to elemental.json.JsonString
	at com.vaadin.flow.server.VaadinService.handleExceptionDuringRequest(VaadinService.java:1589)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1552)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:247)
	... 31 more
Caused by: java.lang.ClassCastException: elemental.json.impl.JreJsonArray cannot be cast to elemental.json.JsonString
	at elemental.json.impl.JreJsonObject.getString(JreJsonObject.java:113)
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.appendNpmBundle(BootstrapHandler.java:895)
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupFrameworkLibraries(BootstrapHandler.java:859)
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupDocumentHead(BootstrapHandler.java:742)
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.getBootstrapPage(BootstrapHandler.java:517)
	at com.vaadin.flow.server.BootstrapHandler.synchronizedHandleRequest(BootstrapHandler.java:458)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1540)
	... 32 more
```

Versions: Vaadin 14.1.2 and Java 8

--- a/flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java
@@ -78,6 +78,7 @@ import com.vaadin.flow.theme.ThemeDefinition;
 import elemental.json.Json;
 import elemental.json.JsonArray;
 import elemental.json.JsonObject;
+import elemental.json.JsonType;
 import elemental.json.JsonValue;
 import elemental.json.impl.JsonUtil;
 
@@ -897,15 +898,21 @@ public class BootstrapHandler extends SynchronizedRequestHandler {
             }
             JsonObject chunks = Json.parse(content);
             for (String key : chunks.keys()) {
+                String chunkName;
+                if(chunks.get(key).getType().equals(JsonType.ARRAY)) {
+                    chunkName = getArrayChunkName(chunks, key);
+                } else {
+                    chunkName = chunks.getString(key);
+                }
                 if (key.endsWith(".es5")) {
                     Element script = createJavaScriptElement(
-                            "./" + VAADIN_MAPPING + chunks.getString(key));
+                            "./" + VAADIN_MAPPING + chunkName);
                     head.appendChild(
                             script.attr("nomodule", true).attr("data-app-id",
                                     context.getUI().getInternals().getAppId()));
                 } else {
                     Element script = createJavaScriptElement(
-                            "./" + VAADIN_MAPPING + chunks.getString(key),
+                            "./" + VAADIN_MAPPING + chunkName,
                             false);
                     head.appendChild(script.attr("type", "module")
                             .attr("data-app-id",
@@ -914,6 +921,18 @@ public class BootstrapHandler extends SynchronizedRequestHandler {
                             .attr("crossorigin", true));
                 }
             }
+        }
+
+        private String getArrayChunkName(JsonObject chunks, String key) {
+            JsonArray chunkArray = chunks.getArray(key);
+
+            for(int i = 0; i <chunkArray.length(); i++) {
+                String chunkName = chunkArray.getString(0);
+                if(chunkName.endsWith(".js")){
+                    return chunkName;
+                }
+            }
+            return "";
         }
 
         private String getClientEngineUrl(BootstrapContext context) {

--- a/flow-server/src/test/java/com/vaadin/flow/server/BootstrapHandlerTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/BootstrapHandlerTest.java
@@ -1,7 +1,7 @@
 package com.vaadin.flow.server;
 
 import javax.servlet.http.HttpServletRequest;
-
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
@@ -1717,6 +1717,39 @@ public class BootstrapHandlerTest {
         Element bundle = scripts.stream()
                 .filter(el -> el.attr("src")
                         .equals("./VAADIN/build/index-1111.cache.js"))
+                .findFirst().get();
+        Assert.assertFalse(bundle.hasAttr("defer"));
+    }
+
+    @Test // #7158
+    public void getBootstrapPage_assetChunksIsAnARRAY_bootstrapParsesOk()
+            throws ServiceException {
+
+        initUI(testUI);
+
+        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+        service.setClassLoader(classLoader);
+
+        String statsJson = "{\n" + " \"errors\": [],\n" + " \"warnings\": [],\n"
+                + " \"assetsByChunkName\": {\n" + "  \"bundle\": [\n"
+                + "    \"build/vaadin-bundle-e77008557c8d410bf0dc.cache.js\",\n"
+                + "    \"build/vaadin-bundle-e77008557c8d410bf0dc.cache.js.map\"\n"
+                + "  ],\n" + "  \"bundle.es5\": [\n"
+                + "    \"build/vaadin-bundle.es5-e71a5a09679e828010c4.cache.js\",\n"
+                + "    \"build/vaadin-bundle.es5-e71a5a09679e828010c4.cache.js.map\"\n"
+                + "  ]" + " }" + "}";
+
+        Mockito.when(classLoader.getResourceAsStream(Mockito.anyString()))
+                .thenReturn(new ByteArrayInputStream(statsJson.getBytes()));
+
+        BootstrapContext bootstrapContext = new BootstrapContext(request, null,
+                session, testUI, this::contextRootRelativePath);
+        Document page = pageBuilder.getBootstrapPage(bootstrapContext);
+
+        Elements scripts = page.head().getElementsByTag("script");
+
+        Element bundle = scripts.stream().filter(el -> el.attr("src")
+                .equals("./VAADIN/build/vaadin-bundle-e77008557c8d410bf0dc.cache.js"))
                 .findFirst().get();
         Assert.assertFalse(bundle.hasAttr("defer"));
     }

